v2/cmd/controller/main.go

```diff
@@ -169,6 +159,33 @@ func main() {
 	}
 }
 
+func getGetDefaultCredential(cfg config.Values, setupLog logr.Logger) (azcore.TokenCredential, error) {
+
+	// If subscriptionID is not supplied, then set default credential to not be used/nil
+	if cfg.SubscriptionID == "" {
+		setupLog.Info("No global credential configured, continuing without.")
+		return nil, nil
+	}
+
+	var credential azcore.TokenCredential
+	var err error
+	if cfg.UseWorkloadIdentityAuth {
+		credential, err = identity.NewWorkloadIdentityCredential(cfg.TenantID, cfg.ClientID)
+		if err != nil {
+			setupLog.Error(err, "unable to get workload identity credential")
+			os.Exit(1)
```

Comment:
IMO do `errors.Wrapf` here, drop the `os.Exit(1)` and return the error. Then where this is called, do:
```
if err != nil { 
    log the error
    os.Exit(1)
}
```

Currently where this method is called you don't do anything with the error, which I think should trip our linters too...